### PR TITLE
chore: moved to hatch build system

### PR DESCRIPTION
**Pull request by russellwmy**
### What problem to solve?
To build python package with a better build system and introduce automated release Github Action
### How to implement?
- Updated config for [hatch](https://github.com/pypa/hatch)
- Added CI for publishing python SDK to test.pypi.org for testing
### How to test?
- Run `pre-commit run --all` to trigger all the checking
- Add release tag with newer version with this branch (feel free to tag, I will delete after merge)
- Check https://github.com/redactive-ai/redactive/actions/workflows/python-sdk.yaml
- Check https://test.pypi.org/project/redactive/

### CODE REVIEW

File: sdks/python/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Redactive Python SDK
+
+The Redactive Python SDK provides a robust and intuitive interface for interacting with the Redactive platform, enabling developers to seamlessly integrate powerful data redaction and anonymization capabilities into their Python applications.
```
Comment (sarafalamaki) [inline sdks/python/README.md:3]: nit: I thought we're using Australian spelling? Anonymisation.

**Review by JJ11teen — APPROVED** [review]
LGTM! Excited to try out hatch!

File: sdks/python/README.md (hunk 1 above)
Comment (russellwmy) [reply sdks/python/README.md:3]: It is generated lol

Comment (russellwmy) [reply sdks/python/README.md:3]: It should be fine. Will update in future PR.

**Pull request merged**